redbox-core/redbox/loader/loaders.py

```diff
@@ -18,15 +28,198 @@
     S3Client = object
 
 
+class Metadata:
+    def __init__(
+        self,
+        env: Settings,
+        s3_client: S3Client,
+        file_name: str,
+    ):
+        self.env = env
+        self.s3_client = s3_client
+        self.metadata = None
+        self.llm = get_chat_llm(self.env, AISettings())
+        self.file_name = file_name
+
+    def get_first_n_tokens(self, chunks: list[dict], n: int) -> str:
+        """From a list of chunks, returns the first n tokens."""
+        current_tokens = 0
+        tokens = ""
+        for chunk in chunks:
+            current_tokens += len(encoding.encode(chunk["text"]))
+            if current_tokens > n:
+                return tokens
+            tokens += chunk["text"]
+        return tokens
+
+    def get_doc_metadata(
+        self, chunks: list[dict], n: int, ignore: list[str] = None
+    ) -> dict[str, Any]:
+        """
+        Use the first n chunks to get metadata using unstructured.
+        Metadata keys in the ignore list will be excluded from the result.
+        """
+        metadata = {}
+        for i, chunk in enumerate(chunks):
+            if i > n:
+                return metadata
+            metadata = self.merge_unstructured_metadata(
+                metadata, chunk["metadata"], ignore
+            )
+        return metadata
+
+    def merge_unstructured_metadata(
```

Comment:
unimportant but is this method static?

Reply:
yes, updated code to mark it as static method.